File: solutions_1-10/Euler_4.py
## find the largest palindrome number made from the product of two digit numbers ? ##

import itertools

canidates = [i for i in xrange(100,1000)]

canidates = [i for i in itertools.combinations(canidates, 2)]

current_high = 0

for canidate in canidates:
    new = canidate[0] * canidate[1]
    if str(new)[::-1] == str(new) and new > current_high:
        current_high = new

print current_high

File: Rust/Rust_solutions_1-10/Euler_4.rs
// Find the largest palindrome number made from the product of two 3-digit 
// numbers?

// Create a generator function - i.e. see what Rust offers
// Does Rust have functional form?

use std::time::Instant;


fn main() {
    let timer = Instant::now();

    let mut answer:u32 = 0;
    for i in 100..999 {
        for j in i..999 {
            let current = i * j;
            let canidate:String = current.to_string();
            let reverse:String = canidate.chars().rev().collect();
            if reverse == canidate && current > answer {
                answer = current;
            }
        }
    }
    println!("Answer: {}", answer);
    println!("Time:   {:?}", timer.elapsed());
}
